src/pages/promo-code/promo-code-edit/test/promo-code-edit.spec.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { Router } from 'react-router-dom';
+import { MockedProvider } from '@apollo/client/testing';
+import { useDispatch } from 'react-redux';
+import { createMemoryHistory } from 'history';
+
+import PromoCodeEdit from '../promo-code-edit';
+import { mocks } from './promo-code-edit.variables';
+
+jest.mock('react-redux');
+
+const mockHistoryPush = jest.fn();
+jest.mock('react-router', () => ({
+  ...jest.requireActual('react-router'),
+  useParams: () => ({
+    id: '61f3ca58fb37d558e8bba8de'
+  }),
+  useHistory: () => ({
+    push: mockHistoryPush
+  })
+}));
+const history = createMemoryHistory();
+
+const dispatch = jest.fn();
+useDispatch.mockImplementation(() => dispatch);
+
+describe('should render component with current data', () => {
+  beforeEach(() => {
+    render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Router history={history}>
+          <PromoCodeEdit />
+        </Router>
+      </MockedProvider>
+    );
+  });
+
+  it('should change promocode code name', async () => {
+    await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
this line duplicates in each test, maybe move it to beforeEach Block?

Reply:
done